bundles/org.openhab.binding.revogi/src/main/java/org/openhab/binding/revogi/internal/RevogiSmartStripControlHandler.java

```diff
@@ -0,0 +1,172 @@
+                switchPlug(command, 6);
+                break;
+            case RevogiSmartStripControlBindingConstants.ALL_PLUGS:
+                switchPlug(command, 0);
+                break;
+            default:
+                logger.info("Sometring went wrong, we've got a message for {}", channelUID.getId());
+        }
+    }
+
+    private void switchPlug(Command command, int port) {
+        if (config == null) {
+            logger.warn("No config available, config object was null");
+            return;
+        }
+        if (command instanceof OnOffType) {
+            int state = convertOnOffTypeToState(command);
+            switchService.switchPort(config.getSerialNumber(), config.ipAddress, port, state);
+        }
+    }
+
+    private int convertOnOffTypeToState(Command command) {
+        if (command == OnOffType.ON) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(RevogiSmartStripControlConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Example for background initialization:
+        scheduler.execute(this::updateStripInformation);
+        Runnable runnable = RevogiSmartStripControlHandler.this::updateStripInformation;
+
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            pollingJob = scheduler.scheduleWithFixedDelay(runnable, 0, config.getPollInterval(), TimeUnit.SECONDS);
+        }
+
+        logger.debug("Finished initializing!");
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    private void updateStripInformation() {
+        if (config == null) {
+            logger.warn("No config available, config object was null");
+            return;
+        }
+        CompletableFuture<Status> futureStatus = statusService.queryStatus(config.getSerialNumber(),
+                config.getIpAddress());
+        futureStatus.thenAccept(this::updatePlugStatus);
+    }
+
+    private void updatePlugStatus(Status status) {
+        if (status.isOnline()) {
+            updateStatus(ThingStatus.ONLINE);
+            handleAllPlugsInformation(status);
+            handleSinglePlugInformation(status);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.GONE,
+                    "Retrieved status code: " + status.getResponseCode());
+        }
+    }
+
+    private void handleSinglePlugInformation(Status status) {
+        for (int i = 0; i < status.getSwitchValue().size(); i++) {
+            int plugNumber = i + 1;
+            updateState("plug" + plugNumber + "#switch", OnOffType.from(status.getSwitchValue().get(i).toString()));
+            updateState("plug" + plugNumber + "#watt", new QuantityType<>(status.getWatt().get(i), MILLI(WATT)));
```

Comment:
Is mW a reasonable default value? You might want to use Watts here, as you also specified one position after the decimal point in the XML.

Reply:
the raw value I receive from the device is milliwatt

Reply:
the change was suggested by @Hilbrand on the old PR

Reply:
You can do that. If I see correctly, the value is displayed in mW in the UI by default. So, I'd suggest to replace the `%unit%` in the XML file by `W` to achieve a reasonable default unit in the UI.

Reply:
> > > * What's the source of the PDF in the doc folder?
> > >   I got this document when I asked the revogi support on how to control this device if I don't want to use their AWS based app.
> 
> Are you sure you may distribute this document? It's also uncommon to include a vendor documentation in the binding. Do you have the ability to store it somewhere else and reference it in the readme?

I thought it was a good idea for tranparency reasons. But anyway, I can include it to my kotlin project and refernence the document here.

Reply:
units are done, too